barman/backup_executor.py

```diff
@@ -358,6 +367,26 @@ def _validate_compression(self, remote_status):
                 exc,
             )
 
+    def validate_backup_args(self, **kwargs):
+        """
+        Validate backup arguments and Postgres configurations. Arguments
+        might be syntactically correct but still be invalid if necessary
+        Postgres configurations are not met.
+
+        :kwparam barman.infofile.LocalBackupInfo parent_backup_info:
+            information of the parent backup in case it is an incremental backup
+            using the postgres mode
+        :raises CommandFailedException: if a command argument is considered invalid
+        """
+
+        if "parent_backup_info" in kwargs:
+            summarize_wal = self.server.postgres.get_setting("summarize_wal")
```

Comment:
this information is gathered, and saved in the backup_info, thanks to @andremagui patch.
doesn't make sense to gather it twice.
On the other hand, @andremagui patch gathers that value during the start backup action which is too late in the chain of calls, we should stop the backup earlier.

Reply:
OK, so @gustabowill I went back and forth in the code a couple of times and I think we should:
* ignore for now that the same check down in the call chain, during the start backup.
* move the implementation into the `BackupManager` object from the `Executor` object. TBH this can be possibly moved into the `Server` object too.
* change the exception from `CommandFailedException` to `BackupException` that is more appropriate IMHO as `CommandFailedException` is used for sub-commands calls like `rsync` or `pg_basebackup` or barman internal sub-calls, and is usually associated to the `Command` object ( https://github.com/EnterpriseDB/barman/blob/f0911baff5372dae1d28dc3d2d8a68aebb2a3310/barman/command_wrappers.py#L115 )

Reply:
Thanks @gcalacoci , I decided to put the logic in the `BackupManager`. I will wait for the #936 to be merged to rebase this PR on top of the master and change what is necessary.